Cauldron/Assets/Scripts/VisitorsProvider.cs

```diff
@@ -0,0 +1,53 @@
+﻿using System.Collections.Generic;
+using CauldronCodebase;
+using UnityEngine;
+
+public class VisitorsProvider
```

Comment:
Also, I have strange convention in this project that "Visitor" is a View (actual animation object that instantiates) and "Villager" is a model (an SO holding some data), so this would be VillagerFamiliarityChecker or something)